app/src/main/java/org/mozilla/fenix/tabstray/ext/TabSessionState.kt

```diff
@@ -4,7 +4,32 @@
 
 package org.mozilla.fenix.tabstray.ext
 
+import androidx.compose.runtime.Composable
 import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.toolbar.MAX_URI_LENGTH
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.compose.inComposePreview
+import org.mozilla.fenix.ext.toShortUrl
+
+/**
+ * Shortens URLs to be more user friendly, by applying [String.toShortUrl]
+ * and making sure it's equal or below the [MAX_URI_LENGTH].
+ */
+@Composable
+fun TabSessionState.toShortUrl(): String {
+    // Truncate to MAX_URI_LENGTH to prevent the UI from locking up for
+    // extremely large URLs such as data URIs or bookmarklets. The same
+    // is done in the toolbar and awesomebar:
+    // https://github.com/mozilla-mobile/fenix/issues/1824
+    // https://github.com/mozilla-mobile/android-components/issues/6985
+    return if (inComposePreview) {
+        "shortenedUrl.com"
```

Comment:
alternatively, it could just be `this.content.url.take(MAX_URI_LENGTH)`, but I don't necessarily feel strongly either way since it's for a preview

Reply:
This also avoids confusion maybe if people start wondering they always just see "shortenedUrl.com" instead of the actual URL